docs/04-resources/04-tutorials/rootstock-vyper.md

```diff
@@ -0,0 +1,222 @@
+    self.favorite_number = new_number
+```
+
+## Deployment Script
+
+Here's our Python script to deploy the contract (`deploy_favorites_unsafe.py`):
+
+```python
+from web3 import Web3
+from dotenv import load_dotenv
+from vyper import compile_code
+import os
+
+load_dotenv()
+
+RPC_URL = os.getenv("RPC_URL")
+
+def main():
+    print("Let's read in the Vyper code and deploy it to the blockchain!")
+    w3 = Web3(Web3.HTTPProvider(RPC_URL))
+    with open("favorites.vy", "r") as favorites_file:
+        favorites_code = favorites_file.read()
+        compliation_details = compile_code(
+            favorites_code, output_formats=["bytecode", "abi"]
+        )
+
+    chain_id = 31  # RSK testnet chain ID
+
+    print("Getting environment variables...")
+    my_address = os.getenv("MY_ADDRESS")
+    private_key = os.getenv("PRIVATE_KEY")
+
+    # Check balance before deployment
+    balance = w3.eth.get_balance(my_address)
+    balance_in_rbtc = w3.from_wei(balance, "ether")
+    print(f"Account balance: {balance_in_rbtc} RBTC")
+
+    if balance == 0:
+        print("Your account has no RBTC! Please get some testnet RBTC from the faucet:")
+        print("1. Go to https://faucet.rsk.co/")
+        print("2. Enter your address:", my_address)
+        print("3. Complete the captcha and request funds")
+        print("4. Wait a few minutes for the transaction to be confirmed")
+        return
+
+    # Create the contract in Python
+    favorites_contract = w3.eth.contract(
+        abi=compliation_details["abi"], bytecode=compliation_details["bytecode"]
+    )
+
+    # Submit the transaction that deploys the contract
+    nonce = w3.eth.get_transaction_count(my_address)
+
+    print("Building the transaction...")
+    transaction = favorites_contract.constructor().build_transaction(
+        {
+            "chainId": chain_id,
+            "from": my_address,
+            "nonce": nonce,
+            "gas": 3000000,  # Higher gas limit for RSK
+            "gasPrice": w3.eth.gas_price * 2,  # Double the gas price to ensure transaction goes through
+        }
+    )
+
+    print("Signing transaction...")
+    signed_txn = w3.eth.account.sign_transaction(transaction, private_key=private_key)
+    print("Deploying contract...")
+    tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)
+    tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)
+    print(f"Contract deployed! Address: {tx_receipt.contractAddress}")
+
+if __name__ == "__main__":
+    main()
+```
+
+## Key Points About RSK Deployment
+
+1. **Chain ID**: RSK testnet uses chain ID 31
+2. **Gas Settings**:
+   - We use a higher gas limit (3,000,000) for RSK
+   - We double the gas price to ensure the transaction goes through
+3. **Transaction Type**:
+   - RSK works best with legacy transactions (using `gasPrice` instead of EIP-1559 parameters)
+
+## Running the Deployment
+
+Execute the deployment script:
+
+```bash
+python deploy_favorites_unsafe.py
```

Comment:
@EdwinLiavaa  Error when running deployment script:

(venv) owanate@Owanates-MacBook-Pro Web3py-Vyper-RootStock % python3 deploy_favorites_unsafe.py
Traceback (most recent call last):
  File "/Users/owanate/Documents/Learning/Web3py-Vyper-RootStock/deploy_favorites_unsafe.py", line 1, in <module>
    from web3 import Web3
ModuleNotFoundError: No module named 'web3'
(venv) owanate@Owanates-MacBook-Pro Web3py-Vyper-RootStock % 


Also specify troubleshooting options too, its usually a good idea to let the reader know how to solve possible errors they may encounter when running the commands.

E.g, If you get this error: ModuleNotFoundError: No module named 'web3'
Do this: